packages/core/components/Puck/index.tsx

```diff
@@ -401,6 +413,32 @@ function PuckLayout<
   );
 
   const dispatch = useAppStore((s) => s.dispatch);
+  const appStoreApi = useAppStoreApi();
+
+  // Add delete hotkey functionality
+  const handleDelete = useCallback(() => {
+    // Access the store state directly to avoid hook call issues in hotkey handlers
+    const storeState = appStoreApi.getState();
+    const selectedItem = storeState.selectedItem;
+    const itemSelector = storeState.state.ui.itemSelector;
+
+    if (selectedItem && itemSelector) {
```

Comment:
This isn't accounting for focus. Right now, if I select a component (Hero), and delete some text in the field (description). This might be fiddly to get right.

https://github.com/user-attachments/assets/3fdb6d8a-f521-4b3e-97bf-d5fab290a80c